OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/AnalyticsProvider.java

```diff
@@ -0,0 +1,327 @@
+        }
+    }
+
+    @Override
+    public void trackShowTranscript(String videoId, Double currentTime,
+                                    String courseId, String unitUrl) {
+        for (Analytics service : services) {
+            service.trackShowTranscript(videoId, currentTime, courseId, unitUrl);
+        }
+    }
+
+    @Override
+    public void trackVideoSeek(String videoId, Double oldTime, Double newTime,
+                               String courseId, String unitUrl, Boolean skipSeek) {
+        for (Analytics service : services) {
+            service.trackVideoSeek(videoId, oldTime, newTime, courseId, unitUrl, skipSeek);
+        }
+    }
+
+    @Override
+    public void trackSingleVideoDownload(String videoId, String courseId, String unitUrl) {
+        for (Analytics service : services) {
+            service.trackSingleVideoDownload(videoId, courseId, unitUrl);
+        }
+    }
+
+    @Override
+    public void trackUserSignUpForAccount() {
+        for (Analytics service : services) {
+            service.trackUserSignUpForAccount();
+        }
+    }
+
+    @Override
+    public void trackCourseOutlineMode(boolean isVideoMode) {
+        for (Analytics service : services) {
+            service.trackCourseOutlineMode(isVideoMode);
+        }
+    }
+
+    @Override
+    public void trackCourseComponentViewed(String blockId, String courseId) {
+        for (Analytics service : services) {
+            service.trackCourseComponentViewed(blockId, courseId);
+        }
+    }
+
+    @Override
+    public void trackDiscoverCoursesClicked() {
+        for (Analytics service : services) {
+            service.trackDiscoverCoursesClicked();
+        }
+    }
+
+    @Override
+    public void trackExploreSubjectsClicked() {
+        for (Analytics service : services) {
+            service.trackExploreSubjectsClicked();
+        }
+    }
+
+    @Override
+    public void trackCreateAccountClicked(String appVersion, String source) {
+        for (Analytics service : services) {
+            service.trackCreateAccountClicked(appVersion, source);
+        }
+    }
+
+    @Override
+    public void trackNotificationTapped(@Nullable String courseId) {
+        for (Analytics service : services) {
+            service.trackNotificationTapped(courseId);
+        }
+    }
+
+    @Override
+    public void trackProfileViewed(@NonNull String username) {
+        for (Analytics service : services) {
+            service.trackProfileViewed(username);
+        }
+    }
+
+    @Override
+    public void trackSectionBulkVideoDownload(String enrollmentId, String section,
+                                              long videoCount) {
+        for (Analytics service : services) {
+            service.trackSectionBulkVideoDownload(enrollmentId, section, videoCount);
+        }
+    }
+
+    @Override
+    public void trackSubSectionBulkVideoDownload(String section, String subSection,
```

Comment:
Nice catch!

At the subsection level we can't bulk download videos, that can only be done on the section level of our Course Outline screen. So, this function should be deleted.

p.s. we have another function `trackSingleVideoDownload` which isn't being called at the sub section level when a download is initiated.